src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalFilterLongReadAlignmentsSAMSpark.java

```diff
@@ -0,0 +1,328 @@
+
+        return filterAndSplitGappedAI(parsedContigAlignments);
+    }
+
+    /**
+     * Idea is to keep mapped contig that has at least two alignments over MQ 20,
+     * or in the case of a single alignment, it must be MQ>20.
+     */
+    private static boolean contigFilter(final AlignedContig contig) {
+        if (contig.alignmentIntervals.size() < 2 ) {
+            return (!contig.alignmentIntervals.isEmpty()) && contig.alignmentIntervals.get(0).mapQual > 20;
+        } else {
+            return contig.alignmentIntervals.stream().mapToInt(ai -> ai.mapQual).filter(mq -> mq > 20).count() > 1;
+        }
+    }
+
+    /**
+     * Manages configuration score-based filtering,
+     * then split the gapped alignments by delegating to {@link GappedAlignmentSplitter}.
+     * Note that this step is essentially a flatMap operation, meaning that one contig may return 1 or multiple contigs:
+     *  *) when 1 contig is yield, it means the contig has only 1 configuration that scored the best
+     *  *) when multiple contigs are yield, it means the contig has several top-scored configurations
+     * How to handle the second scenario can be treated in a separate logic unit.
+     */
+    @VisibleForTesting
+    static JavaRDD<AlignedContig> filterAndSplitGappedAI(final JavaRDD<AlignedContig> parsedContigAlignments) {
+
+        final Set<String> canonicalChromosomes = getCanonicalChromosomes();
+
+        return parsedContigAlignments
+                .mapToPair(alignedContig -> new Tuple2<>(alignedContig.contigName,
+                        new Tuple2<>(alignedContig.contigSequence, pickBestConfigurations(alignedContig, canonicalChromosomes))))
+                .flatMap(InternalFilterLongReadAlignmentsSAMSpark::reConstructContigFromPickedConfiguration)
+                .sortBy(tig -> tig.contigName, true, parsedContigAlignments.getNumPartitions());
+    }
+
+    /**
+     * Pick the best configurations based on a heuristic scoring scheme implemented in
+     * {@link #computeScoreOfConfiguration(List, Set, int)}.
+     * @return a 2-D list, where in the case when multiple configurations are equally top-scored, all such configurations are picked up
+     */
+    @VisibleForTesting
+    static List<List<AlignmentInterval>> pickBestConfigurations(final AlignedContig alignedContig,
+                                                                final Set<String> canonicalChromosomes) {
+
+        // group 1: get max aligner score of mappings to canonical chromosomes and speed up in case of too many mappings
+        final int maxCanonicalChrAlignerScore = alignedContig.alignmentIntervals.stream()
+                .filter(alignmentInterval -> canonicalChromosomes.contains(alignmentInterval.referenceSpan.getContig()))
+                .mapToInt(ai -> ai.alnScore).max().orElse(0); // possible that no mapping to canonical chromosomes
+
+        // speed up if number of alignments is too high (>10)
```

Comment:
Not many of these, but they tend to slow the process down and the power set of these reads' alignment configuration is very large (exponential). Simply removing these reads might not be good because some reads have "a couple of" good alignments and many bad hits to places all over the place. This "speed up" is to remove those bad hits in a brutal way.